python/src/aac/lang/definitions/structural_node.py

```diff
@@ -0,0 +1,60 @@
+"""Provides nodes for decomposing the definition's structure, collating metadata, and providing other QOL improvements."""
+from __future__ import annotations
+from enum import Enum, auto
+from typing import Optional
+from attr import attrib, attrs, validators
+
+import typing
+
+if typing.TYPE_CHECKING:
+    # Have to perform this typing check import to avoid circular dependencies while maintaining type hinting.
+    from aac.lang.definitions.definition import Definition
+
+from aac.lang.definitions.lexeme import Lexeme
+
+
+class NodeType(Enum):
+    """Enumeration values to easily distinguish structural node types."""
+    SCHEMA_STRUCTURE = auto()
+    ENUM_VALUE = auto()
+    ROOT_KEY = auto()
+    LIST_ENTRY = auto()
+    PRIMITIVE_VALUE = auto()
+
+
+@attrs(hash=False, eq=False)
+class StructuralNode:
+    """An Architecture-as-Code definition structural node.
+
+    These nodes are used to build a traversable tree for the definition's structure as opposed to leveraging the
+        unstructured dictionaries, lists, and primitives supplied by just parsing the yaml structure.
+
+    Attributes:
+        node_type (NodeType): Enum value used to quickly distinguish between node types
+        defining_definition (Definition): The definition that defines the structure or enumeration values for the token.
```

Comment:
Why just `Definition` here as the type but is `Optional['Definition']` at line 40?

Reply:
I think Lizz is correct - this should be:

```suggestion
        defining_definition (Optional[Definition]): The `Definition` that defines the structure or enumeration values for the token.
```